pandas/tests/tseries/offsets/test_month.py

```diff
@@ -204,6 +205,56 @@ def test_datetimeindex(self):
             0
         ] == datetime(2012, 1, 31)
 
+    apply_with_extra_offset_cases: _ApplyCases = [
+        (
+            CBMonthEnd(n=1, offset=timedelta(days=5)),
+            {
+                datetime(2021, 3, 1): datetime(2021, 3, 31) + timedelta(days=5),
+                datetime(2021, 4, 17): datetime(2021, 4, 30) + timedelta(days=5),
+            },
+        ),
+        (
+            CBMonthEnd(n=2, offset=timedelta(days=40)),
+            {
+                datetime(2021, 3, 10): datetime(2021, 4, 30) + timedelta(days=40),
+                datetime(2021, 4, 30): datetime(2021, 6, 30) + timedelta(days=40),
+            },
+        ),
+        (
+            CBMonthEnd(n=1, offset=timedelta(days=-5)),
+            {
+                datetime(2021, 3, 1): datetime(2021, 3, 31) - timedelta(days=5),
+                datetime(2021, 4, 11): datetime(2021, 4, 30) - timedelta(days=5),
+            },
+        ),
+        (
+            -2 * CBMonthEnd(n=1, offset=timedelta(days=10)),
+            {
+                datetime(2021, 3, 1): datetime(2021, 1, 29) + timedelta(days=10),
+                datetime(2021, 4, 3): datetime(2021, 2, 26) + timedelta(days=10),
+            },
+        ),
+        (
+            CBMonthEnd(n=0, offset=timedelta(days=1)),
+            {
+                datetime(2021, 3, 2): datetime(2021, 3, 31) + timedelta(days=1),
+                datetime(2021, 4, 1): datetime(2021, 4, 30) + timedelta(days=1),
+            },
+        ),
+        (
+            CBMonthEnd(n=1, holidays=["2021-03-31"], offset=timedelta(days=1)),
+            {
+                datetime(2021, 3, 2): datetime(2021, 3, 30) + timedelta(days=1),
+            },
+        ),
+    ]
```

Comment:
can you just put these directly in the paremeterize?

Reply:
Thanks for the review! Will do. 

As a side note, the existing [tests](https://github.com/pandas-dev/pandas/blob/master/pandas/tests/tseries/offsets/test_month.py#L125-L167) for `.apply` also put the case definition outside of `parametrize` . Happy to quickly change these as well if you would like.